### PR TITLE
Return bool from PresolveImpl's presolve/normalization methods

### PR DESCRIPTION
**Pull request by arcondello**
This way we can indicate if any changes were made. Useful for testing and debugging.

We may want to propagate this to Presolve's user-facing methods as well in the future.

### CODE REVIEW

File: dwave/preprocessing/src/presolveimpl.hpp [hunk 1]
```diff
@@ -53,9 +53,8 @@ class PresolverImpl {
     explicit PresolverImpl(model_type model) : model_(std::move(model)) {}
 
     /// Apply any loaded presolve techniques. Acts on the model() in-place.
-    void apply() {
-        normalize();
-        presolve();
+    bool apply() {
+        return normalize() | presolve();
```
Comment (mdcoury) [inline dwave/preprocessing/src/presolveimpl.hpp:57]: Might make sense to leave a comment here noting that you are intentionally *not* short-circuiting

Comment (mdcoury) [reply dwave/preprocessing/src/presolveimpl.hpp:57]: Alternatively, use the pattern we use below...

File: dwave/preprocessing/src/presolveimpl.hpp [hunk 2]
```diff
@@ -108,23 +112,32 @@ class PresolverImpl {
         if (std::isnan(base.offset())) {
             throw InvalidModelError("biases cannot be NAN");
         }
+
+        return false;
```
Comment (mdcoury) [inline dwave/preprocessing/src/presolveimpl.hpp:116]: Clarifying comment here would be useful for future generations

File: dwave/preprocessing/src/presolveimpl.hpp [hunk 3]
```diff
@@ -108,23 +112,32 @@ class PresolverImpl {
...
+        return changes;
     }
 
     /// Convert a >= constraint into <=.
-    static void normalization_flip_constraint(constraint_type& constraint) {
+    static bool normalization_flip_constraint(constraint_type& constraint) {
         if (constraint.sense() == dimod::Sense::GE) {
             constraint.scale(-1);
+            return true;   // we made changes
+        } else {
```
Comment (mdcoury) [inline dwave/preprocessing/src/presolveimpl.hpp:133]: Else isn't needed here... code seems clearer without it

File: dwave/preprocessing/src/presolveimpl.hpp [hunk 4]
```diff
@@ -163,25 +178,32 @@ class PresolverImpl {
...
     }
 
     /// Remove the offset from a cosntraint. E.g. `x + 1 <= 2` becomes `x <= 1`.
-    static void normalization_remove_offset(constraint_type& constraint) {
+    static bool normalization_remove_offset(constraint_type& constraint) {
         if (constraint.offset()) {
             constraint.set_rhs(constraint.rhs() - constraint.offset());
             constraint.set_offset(0);
+            return true;   // we made changes
+        } else {
```
Comment (mdcoury) [inline dwave/preprocessing/src/presolveimpl.hpp:200]: ditto

File: dwave/preprocessing/src/presolveimpl.hpp (hunk 1 above)
Comment (arcondello) [reply dwave/preprocessing/src/presolveimpl.hpp:58]: Added a comment

**Review by mdcoury — APPROVED** [review]

**Pull request merged**